### DISCUSSION TITLE
Having issue with closing the nested modals using DeviceEventEmitter event listener on iOS

### DISCUSSION THREAD

**Discussion by deepali-jagtap** [react-native-modal/react-native-modal#683] [Q&A]
React native: 0.60.4,
react: 16.8.6,

Currently, in our application, we are showing a popup modal(modal_1) which will show a menu list, and on tap of one of the menu option another popup modal(modal_2) with a sub-menu is shown. So now if the user has modal_1 open and he/she clicks on the modal_1, modal_2 is opened. Now both modals are visible one on top of another. In case of a push notification arrival,  we have to redirect the user to another screen.

We can redirect to the respective screen but are not able to disable both popup modals. Currently, we use the DeviceEventEmitter.emit(event1), where the event listener will set state of both the Modals to false. Now, this disables the modal_2 but we are not able to disable the modal_1. Modal_1 now appears in the frozen state, with the new screen in the background.
   
Any solution/idea on how to fix this issue would be great.

**Comment by ancyrweb** [comment]
It is technically not possible to have two modals open, that's why we are investigating on using portals.
